launcher/src/main/java/org/apache/spark/launcher/SparkSubmitCommandBuilder.java

```diff
@@ -201,8 +205,16 @@
       // - SPARK_DRIVER_MEMORY env variable
       // - SPARK_MEM env variable
       // - default value (512m)
-      String memory = firstNonEmpty(firstNonEmptyValue(SparkLauncher.DRIVER_MEMORY, conf, props),
-        System.getenv("SPARK_DRIVER_MEMORY"), System.getenv("SPARK_MEM"), DEFAULT_MEM);
+      String memory;
+      // Take Thrift Server as daemon
+      if (isThriftServer(mainClass)) {
```

Comment:
To avoid the copy & paste, I'd do:

    String tsMemory = isThriftServer(mainClass) ? getenv("SPARK_DAEMON_MEMORY") : null;
    memory = firstNonEmpty(tsMemory, ...);